website/static/js/index.js

```diff
@@ -8,12 +9,72 @@
 
     var demoInput = document.getElementById("demo-input");
     var demoOutput = document.getElementById("demo-output");
+    var optionsButton = document.getElementById("options");
+    var optionsPanel = document.getElementById("options-panel");
+    var maximize = document.getElementById("maximize");
+    var copyCode = document.getElementById("code");
+    var copyPermalink = document.getElementById("permalink");
+
+    if (window.location.hash === '#demo') {
+        optionsPanel.classList.add('opened');
+        document.body.classList.add("maximized");
+        maximize.innerHTML = 'Minimize editor';
+    }
+
+    var match = window.location.search.match(/[?&]data=([^&]*)/);
+    var data = {};
+    if (match) {
+        try {
+            data = JSON.parse(decodeURIComponent(match[1]));
+        } catch (e) {
+            // eslint-disable-next-line no-console
+            console.warn(e);
+        }
+    }
+
+    if (data.code) {
+        demoInput.value = data.code;
+    }
+
+    var katexOptions = ["displayMode", "throwOnError", "errorColor", "strict",
+        "macros"].map(function(id) {
```

Comment:
I wasn't sure what format was need for the macros, it'd be good to have a couple of pre-populate macros... maybe one that takes an argument.